filebeat/harvester/log.go

```diff
@@ -170,25 +170,34 @@ func (h *Harvester) openFile() (encoding.Encoding, error) {
 	var encoding encoding.Encoding
 
 	f, err := file.ReadOpen(h.Path)
-	if err == nil {
-		// Check we are not following a rabbit hole (symlinks, etc.)
-		if !file.IsRegular(f) {
-			return nil, errors.New("Given file is not a regular file.")
-		}
+	if err != nil {
+		logp.Err("Failed opening %s: %s", h.Path, err)
+		return nil, err
+	}
 
-		encoding, err = h.encoding(f)
-		if err != nil {
+	info, err := f.Stat()
+	if err != nil {
+		logp.Err("Failed getting stats for file %s: %s", h.Path, err)
+		return nil, err
+	}
+	// Compares the stat of the opened file to the state given by the prospector. Abort if not match.
+	if !os.SameFile(h.State.Fileinfo, info) {
+		return nil, errors.New("File info is not identical with opened file. Aborting harvesting and retrying file later again.")
+	}
 
-			if err == transform.ErrShortSrc {
-				logp.Info("Initialising encoding for '%v' failed due to file being too short", f)
-			} else {
-				logp.Err("Initialising encoding for '%v' failed: %v", f, err)
-			}
-			return nil, err
-		}
+	// Check we are not following a rabbit hole (symlinks, etc.)
+	if !file.IsRegular(f) {
```

Comment:
Wouldn't make sense for the new checks to be after the IsRegular check? Not sure if a symlink can cause issues with `Stat` or `Samefile`.